db/db_impl/db_impl_compaction_flush.cc

```diff
@@ -1813,6 +1817,13 @@ Status DBImpl::RunManualCompaction(
     return manual.status;
   }
 
+  // When canceled is not provided by ther user, we assign the pointer of
+  // canceled_internal to it to consolidate canceled and
+  // manual_compaction_paused since DisableManualCompaction() might be called
+  if (manual.canceled == nullptr) {
+    manual.canceled = &manual.canceled_internal;
+  }
```

Comment:
If we make `ManualCompactionState.canceled` a reference too, the assignment should be done in constructor (which I think is cleaner):
```cpp
ManualCompactionState() :
  ...
  canceled(_canceled ? *_canceled : canceled_internal)
```